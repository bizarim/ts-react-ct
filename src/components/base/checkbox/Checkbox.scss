.checkbox-label {
    margin: 0;


    label {
        display: table;
        width: 100%;
    }

    input[type="checkbox"] {
        display: none;
    }

    input[type="checkbox"] {
        + span {
            cursor: pointer;
        }
    }

    input[type="checkbox"] {
        + span {
            span {
                &::before {
                    content: "";
                    position: relative;
                    display: block;
                    width: 20px;
                    height: 20px;
                    border: 1px solid var(--sub-1-border-color);
                    border-radius: 4px;
                    margin: 0;
                }
            }
        }
    }

    input[type="checkbox"]:checked {
        + span {
            span {
                &::before {
                    content: "\2713";
                    font-size: 1rem;
                    color: var(--sub-bg-color);
                    background: var(--main-bg-color);
                    text-align: center;
                }
            }
        }
    }

    .checkbox-el {
        position: relative;
        display: table-cell;
        width: 20px;
        padding-right: 20px;
        vertical-align: middle;
    }

    .checkbox-body {
        cursor: pointer;
        position: relative;
        display: table-cell;
        vertical-align: middle;

        .label-desc {
            margin: 0px;
        }
    }
}
