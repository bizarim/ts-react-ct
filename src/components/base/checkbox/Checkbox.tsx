import React from 'react';

export interface CheckboxProps {
    id: number;
    text: string;
    init: boolean;
    onHandle(id: number, checked: boolean, text: string): void;
}
interface State {
    checked: boolean;
}

export class Checkbox extends React.Component<CheckboxProps, State> {
    public state = {
        checked: this.props.init,
    };
    public render() {
        const { id, text } = this.props;
        const { checked } = this.state;
        return (
            <div id={`${id}`} className="checkbox-label">
                <label>
                    <input
                        type="checkbox"
                        checked={checked}
                        onChange={this.onChange}
                    />
                    <span className="checkbox-el">
                        <span />
                    </span>

                    <span className="checkbox-body">
                        <span className="checkbox-body">
                            <p className="label-desc">{text}</p>
                        </span>
                    </span>
                </label>
            </div>
        );
    }
    private onChange = () => {
        const { id, onHandle, text } = this.props;
        const { checked } = this.state;
        this.setState({ checked: !checked });
        if (onHandle)
            onHandle(id, !checked, text);
    };
}