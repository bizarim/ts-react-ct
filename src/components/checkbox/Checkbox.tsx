import React from 'react';

interface Props {

}
interface State {

}

export class Checkbox extends React.Component<Props, State> {
    public state = {};

    public render() {
        return (
            <div>

            </div>
        );
    }
}