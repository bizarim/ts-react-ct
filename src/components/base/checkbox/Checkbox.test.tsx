import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { CheckboxProps, Checkbox } from './Checkbox';

configure({ adapter: new Adapter() });

const defaultProps: CheckboxProps = {
    id: 1,
    text: 'test',
    init: false,
    onHandle: jest.fn(),
};

const setup = (props: Partial<CheckboxProps> = {}) =>
    shallow(<Checkbox {...{ ...defaultProps, ...props }} />);

describe('Checkbox', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className checkbox-label', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('checkbox-label')).toBeTruthy();
    });

    it('should selected checkbox', () => {
        const onClicked = jest.fn();
        const props: CheckboxProps = {
            id: 1,
            text: 'XXXXX',
            init: false,
            onHandle: onClicked,
        };
        const wrapper = setup(props);
        const checkbox = wrapper.find({ type: 'checkbox' });
        checkbox.simulate('change');
        expect(onClicked).toHaveBeenCalled();

        const text = wrapper.find('p').text();
        expect(text).toBe(props.text);
    });


});
