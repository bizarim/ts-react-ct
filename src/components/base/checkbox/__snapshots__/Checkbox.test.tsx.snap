// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `ShallowWrapper {}`;
